apps/staking/src/api.ts

```diff
@@ -140,96 +139,135 @@ type AccountHistory = {
 }[];
 
 export const getStakeAccounts = async (
-  _connection: Connection,
-  _wallet: AnchorWallet,
-): Promise<StakeAccount[]> => {
-  await new Promise((resolve) => setTimeout(resolve, MOCK_DELAY));
-  return MOCK_STAKE_ACCOUNTS;
+  connection: Connection,
+  wallet: AnchorWallet,
+): Promise<StakeAccountPositions[]> => {
+  const pythStakingClient = new PythStakingClient({ connection, wallet });
```

Comment:
It doesn't technically hold any state, but there are some stuff in the constructor that most likely should not be instantiated on every request.

I agree that the `PythStakingClient` should be instantiated once in the ui and reused, that was the whole purpose of making it a class imo. I didn't do it right now because i was trying to make this PR just to be about the sdk with minimal changes to the ui app.

```
  constructor(config: PythStakingClientConfig) {
    this.connection = config.connection;
    this.wallet = config.wallet;
    this.provider = new AnchorProvider(this.connection, this.wallet, {
      skipPreflight: true,
    });
    this.stakingProgram = new Program(StakingIdl as Staking, this.provider);
    this.integrityPoolProgram = new Program(
      IntegrityPoolIdl as IntegrityPool,
      this.provider
    );
    this.publisherCapsProgram = new Program(
      PublisherCapsIdl as PublisherCaps,
      this.provider
    );
  }
```

Reply:
that's ok @keyvankhademi just merge this as is and I'll handle putting it into context in a follow up PR.